js/app/src/traffic_stats/Queue.svelte

```diff
@@ -0,0 +1,111 @@
+<script lang="ts">
+    import type { QueuePreview, Task } from "./utils";
+    import TaskCard from "./TaskCard.svelte";
+    import { flip } from "svelte/animate";
+
+    export let queue_preview: QueuePreview;
+    export let event_count_per_stage: [number, number, number, number] = [
+        12, 3, 4, 2,
+    ];
+    export let active_workers: number;
+    let active_workers_array = Array.from({ length: active_workers });
+    let flattened_queue: Array<[Task, number, number]>;
+    $: {
+        flattened_queue = [];
+        for (let i = 0; i < queue_preview.length; i++) {
+            for (let j = 0; j < queue_preview[i].length; j++) {
+                flattened_queue.push([queue_preview[i][j], i, j]);
+            }
+        }
+        flattened_queue = flattened_queue;
```

Comment:
😑

Reply:
actually needed to cause Svelte to re-render!